### ISSUE TITLE
N64 Shadows/Textures flickering (Recalbox 4.1 Alpha 2 / Odroid XU4)

### ISSUE THREAD

**Issue by Levingard** [recalbox/recalbox-os#1105]
Hello,
i use Recalbox 4.1 Alpha 2 on an Odroid XU4 and my Problem is, when i start a N64 Game e.g. Mario64, the shadows under the objects and some textures like doorframes are flickering all the time. I use the Standard Settings and changed nothing. I am using Glide64mk2 (others won´t work). I had the same issue on the last release. All other Emulators are working. How can i fix this? i there an Option in the mupen64 config file or something?

**Comment by hutte93** [comment]
same problem here...also using the odroid xu4 port

**Comment by JoargTheBard** [comment]
Anyone figure out the issue?

**Comment by Levingard** [comment]
I tried out Lakka and there ist no Texture Problem with Mupen64 and Glide on my XU4.
But i think the problem is that the support here is very bad and nobody wants to help.

**Comment by substring** [comment]
@Levingard lakka people are good friends of the recalbox team :)

Conerning the bug, this alpha is not the very last build we made of recalbox, gotta try in a newer version

**Comment by phekda** [comment]
> Conerning the bug, this alpha is not the very last build we made of recalbox, gotta try in a newer version

Hi substring! Were can i find this newer build of recalbox for Odroid XU4?
The last build i find is from 08-Oct-2016 and it's version alpha2 the same Levingard mentioned earlier.
I looked everywhere... shouldn't it be available on http://recalbox.remix.free.fr/ normally? 
I can only find version alpha 2 for Odroid XU4... please help because i love N64 games!

**Comment by Levingard** [comment]
Yes, the Alpha 2 seems to be the last one. Where please is the newer one?

**Comment by MastaG** [comment]
I've just built the latest buildroot git and updated the mali driver to r12p0, but the problem persists.

**Comment by substring** [comment]
Most probably an emulator problem

**Comment by MastaG** [comment]
Yes but if the problem doesn't occur on Lakka (which I have not tried yet due to the lack of reicast for arm), then there's probably a patch we need right?

**Comment by rockaddicted** [comment]
are you using mupen64plus (standalone) + gliden64 or glupen64 (libretro core?)

**Comment by MastaG** [comment]
I don't know, I've just followed the wiki to compile recalbox buildroot for xu4 and changed the mali drivers from r7p0 to r12p0, furthermore I've updated reicast so it supports rumble.
However I still have to figure out how to change libretro-cores and/or plugins.
All I did was copying my n64 roms and running them from the gui.

**Comment by rockaddicted** [comment]
Ok, so in ES, `start/game options/advanced/nintendo 64` and switch `emulator` then `core`
I don't have this board, so I can't test.

**Comment by MastaG** [comment]
Ah great, I didn't know it could be setup from the gui.
I'll try different cores/plugins and report back whether it fixes the flickering in Super Mario 64.

**Comment by MastaG** [comment]
So I've tried all combinations and the only two which are working for the XU4 are:
Mupen64plus -> glide64mk2 (seems to be the default for all roms, looks good, fullspeed however flickering shadows)
libretro -> glupen64 (ugly, low-res and only 5fps)

Maybe glide64mk2 has some options config file to fix the shadows?

**Comment by MastaG** [comment]
According to the mupen64plus bugtracker the following values were added to fix this problem:
# If true, use polygon offset values specified below
force_polygon_offset = False
# Specifies a scale factor that is used to create a variable depth offset for each polygon
polygon_offset_factor = 0
# Is multiplied by an implementation-specific value to create a constant depth offset
polygon_offset_units = 0

However I'm not sure what values I should put..

**Comment by hutte93** [comment]
Ok, i've tested some values which have to be in the range from +/- 0.001 to 2
and i got the best results with:

```
force_polygon_offset = True
polygon_offset_factor = -2
polygon_offset_units = -2
```

With these settings the most flickering is gone! :)

Thank you very much for finding these variables!
